src/Channels/ReadableBuffer.cs

```diff
@@ -572,6 +573,115 @@ internal static int FindFirstEqualByteSlow(ref Vector<byte> byteEquals)
             throw new InvalidOperationException();
         }
 
+
+        /// <summary>
+        /// Reads a structure of type T out of a buffer of bytes.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public T ReadBigEndian<[Primitive]T>() where T : struct
```

Comment:
With the new API, both read and write could move to extension methods happily; I wasn't gong to get too excited about  that until we'd nailed the extension method problem